app/models/metric/ci_mixin.rb

```diff
@@ -19,12 +19,17 @@ module Metric::CiMixin
     Metric::LongTermAverages::AVG_METHODS_WITHOUT_OVERHEAD.each do |vcol|
       virtual_column vcol, :type => :float, :uses => :vim_performance_operating_ranges
     end
+  end
 
-    supports :capture do
-      metrics_capture_klass = "#{self.class.module_parent.name}::MetricsCapture".safe_constantize
-      unless metrics_capture_klass&.method_defined?(:perf_collect_metrics)
-        _('This provider does not support metrics collection')
-      end
+  class_methods do
+    def supports_capture?
+      return !unsupported.key?(:capture) if unsupported.key?(:capture)
+
+      klass_name = "#{module_parent.name}::MetricsCapture"
+      return true if klass_name.safe_constantize&.name == klass_name
+
+      unsupported_reason_add(:capture, _('This provider does not support metrics collection'))
+      false
     end
```

Comment:
TODO this doesn't call a per-instance block since it isn't called in the context of the `define_supports_feature_methods`